### PR TITLE
 ocamlPackages.git: 3.3.0 -> 3.3.2; decompress: 1.2.0 -> 1.3.0; duff: 0.3 -> 0.4; and related updates

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

* <https://github.com/mirage/ocaml-git/releases/tag/3.3.1>
* <https://github.com/mirage/ocaml-git/releases/tag/3.3.2>
* <https://github.com/mirage/ocaml-git/releases/tag/carton-v0.2.0>
* <https://github.com/mirage/ocaml-git/releases/tag/carton-v0.4.0>
* <https://github.com/mirage/duff/releases/tag/v0.4>
* <https://github.com/dinosaure/hxd/releases/tag/v0.3.0>
* <https://github.com/dinosaure/hxd/releases/tag/v0.3.1>
* <https://github.com/mirage/decompress/releases/tag/v1.3.0>

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
@ofborg build ocamlPackages.irmin-unix ocamlPackages.imagelib

File: pkgs/development/ocaml-modules/git/unix.nix [hunk 1]
```diff
@@ -15,14 +15,6 @@ buildDunePackage {
 
   useDune2 = true;
 
-  patches = [
-    # https://github.com/mirage/ocaml-git/pull/472
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/git/unix.nix:20]: Please remove fetchpatch from inputs.

File: pkgs/development/ocaml-modules/hxd/default.nix [hunk 2]
```diff
@@ -1,36 +1,37 @@
 { lib, buildDunePackage, fetchurl
-, dune-configurator, cmdliner, angstrom
-, rresult, stdlib-shims, fmt, fpath
+, dune-configurator, cmdliner
+, withLwt ? true, lwt
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/hxd/default.nix:3]:
```suggestion
, ocaml, withLwt ? lib.versionAtLeast ocaml.version "4.07", lwt
```

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/hxd/default.nix:3]: Out of curiosity: What is the issue with 4.06 and lwt?

Comment (vbgl) [reply pkgs/development/ocaml-modules/hxd/default.nix:3]: Syntax error.

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/hxd/default.nix]: Thanks

**Pull request merged**